.github/workflows/rust.yml

```diff
@@ -14,19 +14,18 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@master
         with:
           toolchain: stable
           components: rustfmt
       - uses: Swatinem/rust-cache@v2
-      - name: Get random cache id
-        run: echo "CACHE_ID=$((RANDOM))" >> $GITHUB_ENV
-        shell: bash
-      - uses: pat-s/always-upload-cache@v3.0.11
+      - name: Load rust cache
+        uses: actions/cache@v4
         with:
+          save-always: true
```

Comment:
Fun facts, this doesn't actually work. See https://github.com/actions/cache/issues/1315